### PR TITLE
TVL in various currencies

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

It was requested to display the TVL in various currencies, so this PR introduces these with a random query (since the TVL is fetched once per session).

# Note

This PR does not modify the particular i18n of the display, we keep the custom display of the TVL but updates the symbol according currency.

# Changes

- update did and canister for up-to-date definition of the TVL canister
- implement new option `currency`
- add `Math.random` to worker in order to get once per session a different currency

# Screenshot

gif

### CODE REVIEW

File: frontend/src/lib/canisters/tvl/tvl.did [hunk 1]
```diff
@@ -1,36 +1,31 @@
-type InitArgs = record {
-    governance_id : principal;
-    xrc_id : principal;
-    update_period: nat64;
+// Generated from IC repo commit 3340e9da93c9edd724927ed4fc902b03b0ec6426 'rs/rosetta-api/tvl/tvl.did' by import-candid
+type FiatCurrency = variant {
+    EUR;
+    CNY;
+    JPY;
+    GBP;
```
Comment (bitdivine) [inline frontend/src/lib/canisters/tvl/tvl.did:6]: No USD?

Comment (peterpeterparker) [reply frontend/src/lib/canisters/tvl/tvl.did:6]: No currency = default = USD

**Review by bitdivine — APPROVED** [review]

File: frontend/src/lib/canisters/tvl/tvl.types.ts [hunk 2]
```diff
@@ -1,18 +1,30 @@
 import type { ActorMethod } from "@dfinity/agent";
 import type { Principal } from "@dfinity/principal";
 
-export interface InitArgs {
-  governance_id: Principal;
-  update_period: bigint;
-  xrc_id: Principal;
+export type FiatCurrency =
+  | { CNY: null }
```
Comment (mstrasinskis) [inline frontend/src/lib/canisters/tvl/tvl.types.ts:5]: Not sure, but will it wring anything if we have `undefined` as a option of this type?
`= undefined | { CNY: null }`

**Review by mstrasinskis — APPROVED** [review]
LGTM, thanks. Just a couple of questions.

File: frontend/src/lib/workers/metrics.worker.ts [hunk 3]
```diff
@@ -77,8 +79,32 @@ const syncMetrics = async ({
   syncInProgress = true;
 
   try {
+    const randomCurrency = (): FiatCurrency | undefined => {
+      const random = Math.floor(1 + Math.random() * (5 - 1 + 1));
```
[the remark this answers is not in the document]
Comment (peterpeterparker) [reply frontend/src/lib/workers/metrics.worker.ts]: 😅

Absolutely not, thanks

File: frontend/src/lib/canisters/tvl/tvl.types.ts (hunk 2 above)
Comment (peterpeterparker) [reply frontend/src/lib/canisters/tvl/tvl.types.ts:5]: This type was automatically generated by the candid parser so better not alter it.

But absolutely agree, that would be more clear.

**Pull request merged**